tests/rptest/services/redpanda.py

```diff
@@ -1195,6 +1198,9 @@ def pids(self, node):
     def started_nodes(self):
         return self._started
 
+    def _node_data_dir(self, node):
+        return f"{self.HOST_PERSISTENT_ROOT}/{node.name}"
```

Comment:
IMO this should support the scenario where using multiple partitions is not required (like we had before)

to accomplish this, this logic can be moved to the initialization phase. So having the `PERSISTENT_ROOT` variable change values based on an exposed option (e.g. an environment variable). This option would request the `RedpandaService` class to "use an already-existing subfolder within `PERSISTENT_ROOT` rather than writing on it directly".

This helps having a single place in the code where this logic is handled, rather than spread in multiple places